bin/simple_git_deploy.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+################################################################################
+# Simple deploy script of a BEdita4 instance via git
+#
+# Usage: simple_git_deploy.sh [deploy_user] [app_user] [plugins_list]
+#
+#  * `deploy_user`: user responsible for git/deploy operations, typically file owner
+#  * `app_user`: user launching API actions, typically web server user
+#  * `plugins_list`: comma separated list of active plugins to update together with core
+#
+# Notes:
+#  - when no arguments are passed no deploy or app user is set and all plugins found in `plugins/` are updated
+#  - on single argument only `deploy_user` is set, and used as `app_user`` too
+#  - `plugins_list` use relative path inside `plugins/` for names, example: `BEdita/DevTools,MyPlugin`
+#     on empty `plugins list` all plugins found in `plugins/` are updated
+#
+################################################################################
+
+DEPLOY_USER=''
+DEPLOY_PREFIX_CMD=''
+APP_USER=''
+APP_PREFIX_CMD=''
+
+BE4_DIR=$PWD
+if [ $# -eq 0 ]; then
+    CURR_USR=`whoami`
+    echo "Using current user ($CURR_USR) as deploy and app user"
+fi
+
+if [ ! -z "$1" ]; then
+    DEPLOY_USER="$1"
+    echo "Using $1 as deploy user"
+    DEPLOY_PREFIX_CMD="sudo -H -u $DEPLOY_USER "
+fi
+
+if [ $# -eq 1 ]; then
+    APP_USER="$1"
+    echo "Using $1 as app user too"
+    APP_PREFIX_CMD="sudo -H -u $APP_USER "
+fi
+
+if [ ! -z "$2" ]; then
+    APP_USER="$2"
+    echo "Using $2 as app user"
+    APP_PREFIX_CMD="sudo -H -u $APP_USER "
+fi
+
+# 1. create plugins list read from third arg or look in `plugins/``
+declare -a PLUGINS_DIR
+
+if [ ! -z "$3" ]; then
+    echo "Look for valid plugins in $3"
+    IFS=',' read -a PLUGINS_LIST <<< "$3"
+    for dir in "${PLUGINS_LIST[@]}"; do
+        plug_dir="./plugins/$dir"
+        if [[ -d $plug_dir/.git ]]; then
+            echo "plugin found: $dir"
+            PLUGINS_DIR+=("$plug_dir")
+        else
+            echo "plugin NOT found: $plug_dir"
+        fi
+    done
+else
+    echo "Look for plugins in plugins/ folder"
+    for dir in ./plugins/*/ ; do
```

Comment:
I'm not quite sure that updating _all_ plugins is a desirable behaviour when nothing is specified… 🤔 In development this is probably true, but would it cause problems with current working copy? What happens if a user has changed files in one or more plugin, and they launch this script?

Reply:
That's right, we could change strategy:
* if no argument is passed no plugin is updated
* if a plugin list is specified those plugins are updated
* using a special argument like `--all-plugins` all plugins are updated